### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by GokceGK**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/stackitcloud/terraform-provider-stackit)](https://goreportcard.com/report/github.com/stackitcloud/terraform-provider-stackit) [![GitHub Release](https://img.shields.io/github/v/release/stackitcloud/terraform-provider-stackit)](https://registry.terraform.io/providers/stackitcloud/stackit/latest) ![GitHub go.mod Go version](https://img.shields.io/github/go-mod/go-version/stackitcloud/terraform-provider-stackit) [![GitHub License](https://img.shields.io/github/license/stackitcloud/terraform-provider-stackit)](https://www.apache.org/licenses/LICENSE-2.0)
 
-# Introduction
+# STACKIT Terraform Provider
 
-This project is the official Terraform provider for STACKIT.
+This project is the official [Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs) to integrate all the resources developed by [STACKIT](https://www.stackit.de/en/).
```
Comment (joaopalet) [inline README.md:5]:
```suggestion
This project is the official [Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs) for [STACKIT](https://www.stackit.de/en/), which allows you to manage STACKIT resources through Terraform.

```

Comment (joaopalet) [reply README.md:5]: I think something like this would be a bit clearer, not sure if it still fits the SEO criteria

File: README.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+# STACKIT Terraform Provider
 
-This project is the official Terraform provider for STACKIT.
+This project is the official [Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs) to integrate all the resources developed by [STACKIT](https://www.stackit.de/en/).
         
-# Getting Started
+## Getting Started
 
-To install STACKIT Terraform Provider, copy and paste this code into your Terraform configuration. Then, run `terraform init`.
+To install [STACKIT Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs), copy and paste this code into your Terraform configuration. Then, run `terraform init`.
```
Comment (joaopalet) [inline README.md:9]:
```suggestion
To install the [STACKIT Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs), copy and paste this code into your Terraform configuration. Then, run `terraform init`.

```

File: README.md [hunk 3]
```diff
@@ -49,25 +49,25 @@ When setting up authentication, the provider will always try to use the key flow
...
 
-**Optionally**, you can provide your own private key when creating the service account key, which will then require you to also provide it explicitly to the CLI, additionaly to the service account key. Check the STACKIT Knowledge Base for an [example of how to create your own key-pair](https://docs.stackit.cloud/stackit/en/usage-of-the-service-account-keys-in-stackit-175112464.html#UsageoftheserviceaccountkeysinSTACKIT-CreatinganRSAkey-pair).
+**Optionally**, you can provide your own private key when creating the service account key, which will then require you to also provide it explicitly to the [STACKIT CLI](https://github.com/stackitcloud/stackit-cli/tree/main), additionally to the service account key. Check the STACKIT Knowledge Base for an [example of how to create your own key-pair](https://docs.stackit.cloud/stackit/en/usage-of-the-service-account-keys-in-stackit-175112464.html#UsageoftheserviceaccountkeysinSTACKIT-CreatinganRSAkey-pair).
```
Comment (joaopalet) [inline README.md:60]: Oh, I think this should have been `provide it explicitly to the Terraform Provider` instead, not the CLI

File: README.md [hunk 4]
```diff
@@ -49,25 +49,25 @@ When setting up authentication, the provider will always try to use the key flow
...
-## Key flow
+### Key flow
 
-    The following instructions assume that you have created a service account and assigned it the necessary permissions, e.g. `project.owner`.
+    The following instructions assume that you have created a service account and assigned the necessary permissions to it, e.g. `project.owner`.
 
 To use the key flow, you need to have a service account key, which must have an RSA key-pair attached to it.
 
-When creating the service account key, a new pair can be created automatically, which will be included in the service account key. This will make it much easier to configure the key flow authentication in the CLI, by just providing the service account key.
+When creating the service account key, a new pair can be created automatically, which will be included in the service account key. This will make it much easier to configure the key flow authentication in the [STACKIT CLI](https://github.com/stackitcloud/stackit-cli/tree/main), by just providing the service account key.
```
Comment (joaopalet) [inline README.md:58]: Same here

File: README.md (hunk 1 above)
Comment (GokceGK) [reply README.md:5]: It still has the links and 3 key words STACKIT, Terraform, Provider :) So it looks good to me

File: README.md [hunk 5]
```diff
@@ -154,26 +156,38 @@ Additionally:
...
-# Migration
+## Migration
 
 For guidance on how to migrate to using this provider, please see our [Migration Guide](./MIGRATION.md).
 
-# Reporting Issues
+## Reporting Issues
 
-If you encounter any issues or have suggestions for improvements, please open an issue in the repository.
+If you encounter any issues or have suggestions for improvements, please open an issue in the [repository](https://github.com/SchwarzIT/terraform-provider-stackit/issues).
```
Comment (DiogoFerrao) [inline README.md:173]: This is the link to the _other_ provider 👀. This is the correct one:
https://github.com/stackitcloud/terraform-provider-stackit/issues

Comment (GokceGK) [reply README.md]: thx a lot :) changed the link

**Review by joaopalet — APPROVED** [review]

**Pull request merged**